File: app/src/main/java/com/auto/billiardnote/ui/home/DrawingButton.java
package com.auto.billiardnote.ui.home;

import android.content.Context;
import android.graphics.Color;
import android.util.AttributeSet;

import com.auto.billiardnote.ui.home.draw.DrawingTool;

public class DrawingButton extends androidx.appcompat.widget.AppCompatImageButton {

    private DrawingTool tool;

    public DrawingButton(Context context) {
        super(context);
    }

    public DrawingButton(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public DrawingButton(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public DrawingTool getTool() {
        return tool;
    }

    public void setTool(DrawingTool tool) {
        this.tool = tool;
    }

    public void setModeChange(boolean enable) {
        this.setEnabled(enable);
        int enableColor = Color.WHITE;
        int disableColor = Color.DKGRAY;
        setBackgroundColor(enable ? enableColor : disableColor);
    }
}

File: app/src/main/java/com/auto/billiardnote/ui/home/DrawingButton.kt
package com.auto.billiardnote.ui.home

import android.content.Context
import android.graphics.Color
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatImageButton
import com.auto.billiardnote.ui.home.draw.DrawingTool

class DrawingButton : AppCompatImageButton {
    var tool: DrawingTool? = null

    constructor(context: Context?) : super(context!!) {}
    constructor(context: Context?, attrs: AttributeSet?) : super(context!!, attrs) {}
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context!!, attrs, defStyleAttr
    ) {
    }

    override fun setEnabled(enabled: Boolean) {
        super.setEnabled(enabled)
//        children.forEach { it.isEnabled = enabled }
    }

    fun setModeChange(enable: Boolean) {
        isEnabled = enable
        val enableColor = Color.WHITE
        val disableColor = Color.DKGRAY
        setBackgroundColor(if (enable) enableColor else disableColor)
    }
}